### DISCUSSION TITLE
[Question]: How to ensure LibreChat uses proxy?

### DISCUSSION THREAD

**Discussion by igodsewer** [danny-avila/LibreChat#1378] [Q&A] [answered]
### What is your question?

Hi, everyone!
Actually, two questions here.

First one - which is correct format for PROXY entry in configuration file? I assumed usual `http://proxy:port`, am I correct?
Second one - How can I make sure LibreChat is running all requests to endpoints through a proxy server entered in .env file?

My ENDPOINTS section in .env file is as follows (everything else is on default settings - i'm in a test environment so its fine :P):

```
#===================================================#
#                     Endpoints                     #
#===================================================#

ENDPOINTS=openAI
PROXY=http://1.2.3.4:1234
```

When I tried going into LibreChat container and curl-ing https://ifconfig.co with `curl -v https://ifconfig.co`, it responded with my non-proxied IP address, so I guess proxying is done on the app level.

My proxy server is a simple Squid proxy server hosted on GCP, confirmed working from my local PC.

Thanks in advance!

### More Details

described above

### What is the main subject of your question?

Endpoints, Other

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
the PROXY setting is only intended for proxying requests to the AI provider endpoints, not for the entire app/container. Setting it should use your proxy for requests to OpenAI servers.

If you are expecting to be using your proxy from the container, this is unrelated to LibreChat and would need to be a docker configuration on your part.

Note not all providers support an explicit proxy setting, namely Google. In this case, a global setting is necessary, and may also address your concern of proxying all server requests: https://github.com/danny-avila/LibreChat/issues/1372

This is not yet developed and is not a big priority but will get around to it eventually. PRs welcome

**Reply by danny-avila** [reply]
The Docker configuration solution might be your best bet, since I'm sure someone has needed this before and figured out to simply proxy all outgoing requests from the container.

**Reply by igodsewer** [reply]
Hello and thanks for your reply!

I'm not really familiar with LibreChat architecture, that curl command was just a blind suggestion on how to check if everything is working without digging in too deep.
 
How can I double check that requests to endpoints are _actually being_ proxied? And is my format of `PROXY=` entry correct?

Thanks!

**Reply by igodsewer** [reply]
Oh, I'm so sorry - I forgot to mention. I use docker-compose setup on Ubuntu 22.04, nothing really special.

**Reply by danny-avila** [reply, marked as answer]
The format looks correct. Look into mitmproxy. It's not super user friendly but this is what I use to test that the proxy is actually pinged.

**Reply by igodsewer** [reply]
Great, thanks! 
I think, I'll just look up some packets via tcpdump and if I see stuff flowing through proxy server, it'll be just fine for now.
